pkg/registry/flowcontrol/rest/prioritylevelconfiguration.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rest
+
+import (
+	"context"
+	"fmt"
+
+	flowcontrolv1beta1 "k8s.io/api/flowcontrol/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	flowcontrolclient "k8s.io/client-go/kubernetes/typed/flowcontrol/v1beta1"
+	"k8s.io/klog/v2"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+type plEnsureStrategy interface {
```

Comment:
There is a lot of copy-paste between these two files.

I'm wondering if it wouldn't be possible to refactor it to share the code.

If we would change everything to "runtime.Object" then it seems doable
[you just parametrize it with type-based functions, e.g.:

```
  ..
  createFunc func(runtime.Object) error

...

  createFunc: func(obj runtime.Object) error {
     pl, ok := obj.(*PriorityLevel)
      if !ok { .. }
     return client.PriorityLevels().Create(context.TODO(), pl, ...)
  }
```

There is a bit of boilerplate, but you will have the same share logic.

Reply:
@wojtek-t  I initially thought about this but backed away, was not sure if I would get support on it. :) Thanks for the suggestion, I managed to isolate all the boilerplate/type specific code so it's easy to read the shared code. Let me know if this looks good to you.